torch/_inductor/codegen/cpp_wrapper_cpu.py

```diff
@@ -1580,6 +1580,12 @@ def make_allocation(
         self, name, device, dtype, shape, stride, buffer_if_can_stack_allocate=None
     ):
         orig_stride = stride
+        # This is a fix for test_duplicated_params_abi_compatible_cpu_with_stack_allocation_and_minimal_arrayref_interface
+        # The https://github.com/pytorch/pytorch/pull/130956 may convert a symint stride to int,
+        # which generates a const int arg rather than a static constexpr int arg.
+        for i, item in enumerate(orig_stride):
+            if isinstance(item, int):
+                orig_stride[i] = sympy.Integer(item)
```

Comment:
This is weird. Shouldn't https://github.com/pytorch/pytorch/blob/de35d3062f4d1404296b379c6899e6350eb9970e/torch/_inductor/codegen/cpp_wrapper_cpu.py#L1583 this return True in this case, and then generate [static constexpr](https://github.com/pytorch/pytorch/blob/de35d3062f4d1404296b379c6899e6350eb9970e/torch/_inductor/codegen/cpp_wrapper_cpu.py#L1550C36-L1550C52)?

Reply:
no.
https://github.com/pytorch/pytorch/blob/5f5a93c3ee7f33d7a18ec5c2d9f5cea1e44c2192/torch/fx/experimental/symbolic_shapes.py#L4458
the above line `self.simplify(expr) ` can't handle `int` type. It expects a symbol expression.

Reply:
Well, it's not hard to catch that in `statically_known_int_or_none`, right?

Reply:
I'm not sure if this will influence other paths. but yes, let's try fixed in statically_known_int_or_none https://github.com/pytorch/pytorch/pull/130956/commits/fa23a62bce20f31c58045aeb09c8d92086c33edd